graphql-native/src/main/java/io/ballerina/stdlib/graphql/runtime/engine/EngineUtils.java

```diff
@@ -107,14 +110,18 @@ private EngineUtils() {
     // Native Data Fields
     public static final String GRAPHQL_SERVICE_OBJECT = "graphql.service.object";
 
-    static BMap<BString, Object> getErrorDetailRecord(BError error, BObject node) {
+    static BMap<BString, Object> getErrorDetailRecord(BError error, BObject node, List<Object> pathSegments) {
         BMap<BString, Object> location = node.getMapValue(LOCATION_FIELD);
         ArrayType locationsArrayType = TypeCreator.createArrayType(location.getType());
         BArray locations = ValueCreator.createArrayValue(locationsArrayType);
         locations.append(location);
+        ArrayType segmentArrayType = TypeCreator
```

Comment:
Wouldn't this be a better name?

```suggestion
        ArrayType pathSegmentArrayType = TypeCreator
```